### PR TITLE
[e2e] Switch ethereum chain multiple dapps

### PR DESCRIPTION
**Pull request by tmashuang**
## **Description**
_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

Closes https://github.com/MetaMask/MetaMask-planning/issues/1432
Closes https://github.com/MetaMask/MetaMask-planning/issues/911

E2E tests for switchEthereumChain for multiple dapps and notification queueing for those dapps.

## **Manual testing steps**

_1. Pull the branch down
_2. yarn && yarn build:test && yarn test:e2e:chrome:rpc
_3. Tests should pass

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

**Comment by adonesky1** [comment]
@tmashuang is there some trick to getting this test to run? The test is currently failing for me with:
```
<testsuite name="Mocha Tests" tests="4" failures="0" errors="4" skipped="0" timestamp="Mon, 16 Oct 2023 17:24:17 GMT" time="29.586">
<testcase classname="Switch Ethereum Chain for two dapps" name="switches the chainId of two dapps when switchEthereumChain of one dapp is confirmed" time="29.1"><failure>waitUntilXWindowHandles timed out polling window handles
Error: waitUntilXWindowHandles timed out polling window handles
    at Driver.initialBalanceInHex [as waitUntilXWindowHandles] (test/e2e/webdriver/driver.js:408:3)
    at windowHandles (test/e2e/helpers.js:712:28)
    at /metamask/extension/test/e2e/json-rpc/switchEthereumChain.spec.js:38:15
    at withFixtures (test/e2e/helpers.js:107:9)
    at Context.&#x3C;anonymous&#x3E; (test/e2e/json-rpc/switchEthereumChain.spec.js:3:3)</failure></testcase>
    ```

**Comment by tmashuang** [comment]
@adonesky1 Could you run `yarn && yarn build:test && yarn test:e2e:single test/e2e/json-rpc/switchEthereumChain.spec.js --browser=chrome --leave-running=true` and see where it's hanging for you?

**Comment by adonesky1** [comment]
Hey @tmashuang yeah that's exactly the path I took where I was getting those errors... 🤔

File: test/e2e/json-rpc/switchEthereumChain.spec.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        // Window handles should only be expanded mm, dapp one, dapp 2 (3 total)
+        await driver.wait(async () => {
+          windowHandles = await driver.getAllWindowHandles();
+          return windowHandles.length === 3;
+        });
+      },
+    );
+  });
+
+  it('should queue send tx after switchEthereum request with a warning, denying presents pending tx', async function () {
```
Comment (adonesky1) [inline test/e2e/json-rpc/switchEthereumChain.spec.js:234]:
```suggestion
  it('should queue send tx after switchEthereum request with a warning, if switchEthereum request is cancelled should show pending tx', async function () {
```

**Review by adonesky1 — DISMISSED** [review]
Finally got this working. LGTM! Just a couple of non-blocking nits

**Review by adonesky1 — APPROVED** [review]
LGTM!

File: test/e2e/fixture-builder.js [hunk 2]
```diff
@@ -511,6 +511,29 @@ class FixtureBuilder {
...
+        networkConfigurationId: {
+          chainId: CHAIN_IDS.LOCALHOST,
+          nickname: 'Localhost 8545',
+          rpcPrefs: {},
+          rpcUrl: 'http://localhost:8545',
+          ticker: 'ETH',
+          networkConfigurationId: 'networkConfigurationId',
+        },
+        '76e9cd59-d8e2-47e7-b369-9c205ccb602c': {
+          id: '76e9cd59-d8e2-47e7-b369-9c205ccb602c',
```
Comment (jiexi) [inline test/e2e/fixture-builder.js:526]: `networkConfigurationId2` or something like that? no strong preference, just wondering if uuid is necessary

**Review by jiexi — APPROVED** [review]
LGTM. feel free to ignore the nit

**Pull request merged**